src/screens/Server.js

```diff
@@ -69,10 +72,20 @@ class Server extends React.Component {
     }
 
     this.setState({ loading: true, errorMessage: '' });
+    const currentNetwork = hathorLib.network.getNetwork().name;
+    const currentServer = hathorLib.helpers.getServerURL();
     // Update new server in local storage
     hathorLib.wallet.changeServer(newServer);
     const promise = version.checkApiVersion();
-    promise.then(() => {
+    promise.then((data) => {
+      // If the user has not explicitly said that he wants to connect to a testnet but end up connecting to one, we show error and don't continue
+      if (!this.state.testnetServer && data.network !== 'mainnet') {
+        this.setState({ errorMessage: t`You tried to connect to a testnet server and did not select that. Beware if someone asked you to do it, the tokens from testnet have no value.`, loading: false });
```

Comment:
Isn't it better to open a modal when the user switches to testnet asking him to confirm? I feel that having the checkbox there always looks weird and a modal explicitly states what the action is. We can even make the default action on the modal be to cancel it.

Reply:
To be honest I feel that the modal or the current checkbox is pretty much the same, I even prefer the checkbox but I see no benefits in changing to add the modal.

Reply:
The modal would only open if the user is switching to a testnet. It looks weird to me having the checkbox if the user hasn't typed a testnet node api.

Reply:
I think this screen will be used either by experienced users who knows what they are doing, so the checkbox would be ok and by users that are being deceived and in this case I prefer having this protection before changing the server, rather than creating a modal that can't be closed and has some requirements to protect this user who has already done something wrong. But again, it's pretty much the same for me, I don't see much difference, I can change to a modal.

Reply:
I think a modal where the default action is cancelling the change seems safer. Or even requiring on the modal to select a checkbox to confirm. We should get more opinions @msbrogli @jansegre @andreabadesso @thr0wn

Reply:
If we go with a modal, I think we should require the user to write "I want to connect to the testnet" or similar (instead of just clicking in a button). Anyway, I think connecting to the testnet is a major thing and should be "hard to do". We can require the user to enable "Dev mode" to do that, for example. I know it feels late to discuss UI here but I feel that the easiest it is, the more people will be scammed.

Reply:
I would agree with the modal if changing the server was "routine usage", but changing the server is already an "advanced" usage of the wallet, there are multiple steps before being able to do it.

So I think the checkbox should be enough.

Reply:
I would agree with the modal if changing the server was "routine usage", but changing the server is already an "advanced" usage of the wallet, there are multiple steps before being able to do it.

So I think the checkbox should be enough.

Reply:
I just saw the testnet-connection.mov video and I think that's the best one. I agree with @obiyankenobi that we should rename the buttons to "Connect to testnet" and "Cancel change" while cancelling is the default action.

We can go further and ask users to write "testnet" in a textbox to force them to read before clicking in the button. I feel it is a serious business to allow users to connect to the testnet from our wallet because it opens a new world of opportunities for scammers.

Reply:
I've just added the latest video with the updated UI/UX. We can track the comments in this thread.

@andreabadesso @obiyankenobi @msbrogli

Reply:
This flow seems really good to me.

Reply:
Looks good to me

Reply:
Great. \o/